### DISCUSSION TITLE
running ./build_openroad.sh -o -l --clean-force does not properly cleanup the directory tools/yosys/abc

### DISCUSSION THREAD

**Discussion by gkamendje** [The-OpenROAD-Project/OpenROAD-flow-scripts#1192] [Ideas]
The following code in  `./build_openroad.sh` does not clean up the directory `tools/yosys/abc`
```
__cleanup()
{
        if [ ! -z "${CLEAN_FORCE+x}" ]; then
                CLEAN_CMD="-x -d --force"
        else
                CLEAN_CMD="-x -d --interactive"
        fi
        echo "[INFO FLW-0026] Cleaning up previous binaries and build files."
        git clean ${CLEAN_CMD} tools
        git submodule foreach --recursive git clean ${CLEAN_CMD}
}
```
It turns out that running `./build_openroad.sh -o -l --clean-force` will not remove `abc` related binaries compiled with a previous libraries dependency set. In my case for example, I had previously compiled in Ubuntu 20.04 which comes with  `libreadline.so.7`. After upgrading my system to Ubuntu 22.04, despite re-building ORFS with `--clean-force`, `abc` would still show up as linked to   `libreadline.so.7` 
```
ldd abc-bafd2a7
        linux-vdso.so.1 (0x00007ffc891db000)
        libreadline.so.7 => not found
```
instead of  `libreadline.so.8` simply because the directory `tools/yosys/abc` had not been cleaned up.

**Comment by maliberty** [comment]
@vvbandeira please look at the cleanup issue

**Comment by vvbandeira** [comment]
Thanks for pointing this out @gkamendje. Turns out that `abc` inside `tools/yosys` is not registered as a submodule, so the current commands do not clean it correctly. PR #1195 should address it, can you confirm?

**Reply by vijayank88** [reply]
@gkamendje Have you updated the commit and is resolved?

**Reply by gkamendje** [reply]
@vvbandeira the fix currently deletes only the abc binary. Would it make sense to wipe out the entire directory and let Yosys recreate it?

**Reply by vvbandeira** [reply]
It will clean the workspace, so it should include object files as well. Are you still having problems changing the ABC version?

**Reply by gkamendje** [reply]
@vvbandeira Everything is fine thanks
